### PR TITLE
Sort listings (returned by `GET /listings`) by `updatedAt`

### PR DESCRIPTION
**Pull request by anders-schneider**
## Issue

- Addresses #193

## Description

This change orders the results of the backend `GET /listings` by which was most recently updated. It follows "Option C" from https://github.com/CityOfDetroit/bloom/issues/193#issuecomment-915427099. And it delegates the work of creating the `OrderByCondition` to a new `getOrderByCondition` function, since the logic in there will become more complex.

This change also rationalizes the pre-existing sorting behavior, which wasn't intended to and didn't make sense. The pre-existing behavior:
- Listings would be sorted globally (ie across all pages, if paginated) by a meaningless Listing ID
- Within a page, listings would be sorted by applicationDueDate, then by applicationOpenDate, then by maxOccupancy.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

```
$ cd backend/core
$ yarn test
$ yarn test:e2e:local
```

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes
- [ ] I have run `yarn generate:client` if I made backend changes

### CODE REVIEW

File: backend/core/test/listings/listings.e2e-spec.ts [hunk 1]
```diff
@@ -242,6 +242,17 @@ describe("Listings", () => {
...
   })
 
+  it("should sort results from most recently updated to least", async () => {
+    const res = await supertest(app.getHttpServer()).get(`/listings?limit=all`).expect(200)
+    for (let i = 0; i < res.body.items.length - 1; ++i) {
+      const currentUpdatedAt = new Date(res.body.items[i].updatedAt)
+      const nextUpdatedAt = new Date(res.body.items[i + 1].updatedAt)
+
+      // Verify that each listing's updatedAt timestamp is more recent than the next listing's.
+      expect(currentUpdatedAt.getTime()).toBeGreaterThan(nextUpdatedAt.getTime())
```
Comment (avaleske) [inline backend/core/test/listings/listings.e2e-spec.ts:252]: Is it worth making another request in this test for the next page, to make sure results are ordered correctly across pages, or does the unit test in listings.spec.ts cover that well enough?

**Review by avaleske — APPROVED** [review]
LGTM!

**Comment by avaleske** [comment]
Could you also contribute this upstream, just so they have smaller things to review and so they know we're working on sorting?

File: backend/core/test/listings/listings.e2e-spec.ts (hunk 1 above)
Comment (anders-schneider) [reply backend/core/test/listings/listings.e2e-spec.ts:252]: I like this! Added a second test for the interaction of pagination and sorting

**Comment by anders-schneider** [comment]
> Could you also contribute this upstream, just so they have smaller things to review and so they know we're working on sorting?

Will do!

**Comment by willrlin** [comment]
FYI, I think this may have unintentionally caused the unit summaries to not be sorted by bedroom type. On the deployed site, I'm seeing some properties list in the order "4br, 3br, 2br" and others "2br, studio, 1br".

**Comment by anders-schneider** [comment]
> FYI, I think this may have unintentionally caused the unit summaries to not be sorted by bedroom type. On the deployed site, I'm seeing some properties list in the order "4br, 3br, 2br" and others "2br, studio, 1br".

Thanks for the heads up - I've got a fix for this and will send a PR soon

**Comment by anders-schneider** [comment]
https://github.com/CityOfDetroit/bloom/pull/547 is out to address the misordering of unit summaries.

**Pull request merged**